services/dynamic-sidecar/tests/unit/test_api_workflow_service_metrics.py

```diff
@@ -0,0 +1,454 @@
+from aiodocker.utils import clean_filters
+from aiodocker.volumes import DockerVolume
+from asgi_lifespan import LifespanManager
+from fastapi import FastAPI
+from httpx import AsyncClient
+from models_library.generated_models.docker_rest_api import ContainerState
+from models_library.generated_models.docker_rest_api import Status2 as ContainerStatus
+from models_library.rabbitmq_messages import (
+    RabbitResourceTrackingHeartbeatMessage,
+    RabbitResourceTrackingMessages,
+    RabbitResourceTrackingStartedMessage,
+    RabbitResourceTrackingStoppedMessage,
+    SimcorePlatformStatus,
+)
+from models_library.services_creation import CreateServiceMetricsAdditionalParams
+from pydantic import AnyHttpUrl, parse_obj_as
+from pytest_mock import MockerFixture
+from pytest_simcore.helpers.utils_envs import EnvVarsDict, setenvs_from_dict
+from servicelib.fastapi.long_running_tasks.client import (
+    Client,
+    TaskClientResultError,
+    TaskId,
+    periodic_task_result,
+)
+from servicelib.fastapi.long_running_tasks.client import setup as client_setup
+from simcore_service_dynamic_sidecar._meta import API_VTAG
+from simcore_service_dynamic_sidecar.core.docker_utils import get_container_states
+from simcore_service_dynamic_sidecar.models.schemas.containers import ContainersCreate
+from simcore_service_dynamic_sidecar.models.shared_store import SharedStore
+from tenacity import AsyncRetrying, TryAgain
+from tenacity.stop import stop_after_delay
+from tenacity.wait import wait_fixed
+
+_FAST_STATUS_POLL: Final[float] = 0.1
+_CREATE_SERVICE_CONTAINERS_TIMEOUT: Final[float] = 60
+_BASE_HEART_BEAT_INTERVAL: Final[float] = 0.1
+
+
+@pytest.fixture(params=[1, 2])
+def container_names(request: pytest.FixtureRequest) -> list[str]:
+    return [f"service-{i}" for i in range(request.param)]
+
+
+@pytest.fixture
+def raw_compose_spec(container_names: list[str]) -> dict[str, Any]:
+    base_spec: dict[str, Any] = {"version": "3", "services": {}}
+
+    for container_name in container_names:
+        base_spec["services"][container_name] = {
+            "image": "alpine:latest",
+            "command": ["sh", "-c", "sleep 100000"],
+        }
+
+    return base_spec
+
+
+@pytest.fixture
+def compose_spec(raw_compose_spec: dict[str, Any]) -> str:
+    return json.dumps(raw_compose_spec)
+
+
+@pytest.fixture
+def backend_url() -> AnyHttpUrl:
+    return parse_obj_as(AnyHttpUrl, "http://backgroud.testserver.io")
+
+
+@pytest.fixture
+def mock_environment(
+    mock_core_rabbitmq: dict[str, AsyncMock],
+    monkeypatch: pytest.MonkeyPatch,
+    mock_environment: EnvVarsDict,
+) -> EnvVarsDict:
+    setenvs_from_dict(
+        monkeypatch,
+        {
+            "RESOURCE_TRACKING_HEARTBEAT_INTERVAL": f"{_BASE_HEART_BEAT_INTERVAL}",
+            "RABBIT_HOST": "mocked_host",
+            "RABBIT_SECURE": "false",
+            "RABBIT_USER": "mocked_user",
+            "RABBIT_PASSWORD": "mocked_password",
+        },
```

Comment:
I see you duplicate this in every one of your tests. why not putting this in your base mock_environment fixture instead and only change it if needed?>